File: app/src/main/java/io/ipoli/android/app/ui/calendar/CalendarAdapter.java
package io.ipoli.android.app.ui.calendar;

import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import io.ipoli.android.app.utils.Time;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 2/20/16.
 */
public interface CalendarAdapter<E extends CalendarEvent> {
    List<E> getEvents();

    View getView(ViewGroup parent, int position);

    void onStartTimeUpdated(E calendarEvent, int oldStartTime);

    void notifyDataSetChanged();

    void updateEvents(List<E> calendarEvents);

    void onDragStarted(View dragView, Time time);

    void onDragEnded(View dragView);

    void removeEvent(E calendarEvent);

    void onDragMoved(View dragView, Time time);
}

File: app/src/main/java/io/ipoli/android/quest/calendar/ui/CalendarAdapter.kt
package io.ipoli.android.quest.calendar.ui

import android.content.Context
import android.widget.ArrayAdapter

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 9/2/17.
 */
abstract class CalendarAdapter<T : CalendarEvent>(context: Context, resource: Int, events: List<T>) :
    ArrayAdapter<T>(context, resource, events) {
    protected val resource = resource
    abstract fun onStartEdit(position: Int)
    abstract fun onStopEdit(position: Int)
}